### PR TITLE
Add colors

### PR DESCRIPTION
**Pull request by gagelarsen**
**Primary changes in this Pull Request:**

- Add github actions example for this repo
- Cleanup some extra flake options
- Add the options to color each column
- Update tests to use pytest fixtures

### CODE REVIEW

**Comment by gagelarsen** [comment]
Here is an example output

pretty_table_screenshot

**Comment by gagelarsen** [comment]
What did you decide about this?

**Comment by Justintime50** [comment]
Honestly just haven't had time yet haha. One last item that would be interesting that's related - could we add conditional coloring? For instance, let's say I wanted to include this in a project that displayed if something was included or not, truthy answers get a green color while falsy answers get a red color (or for something like 'x' and '-')?

Let me know, then I can take another look.

**Comment by gagelarsen** [comment]
Can you give me an ascii table example? I'm not sure exactly what you mean, but i'm sure i can do it.

**Comment by Justintime50** [comment]
Certainly. Take the following example:

```
| ID | Search Term | File  | Contains Search Term |
| 1   | string1           | file1 | Yes                                |
| 2  | string2           | file2 | No                                 |
| 3  | string3           | file3 | Yes                               |
```
**Disregard the terrible formatting. Silly GitHub.**

Let's say I want to only color code the last column, yes's get the color green while no's get the color red. Ideally it would be nice if it could be conditionally specified (color, the string to color, etc)

**Comment by gagelarsen** [comment]
Okay, what do you think of this?

```python
headers = ['ID', 'Name', 'Occupation', 'Employed']
rows = [
    [1, 'Justin', 'Software Engineer', True],
    [2, 'Misty', 'Receptionist', False],
    [3, 'John', None, False],
]
```
```python
formatted_table = PrettyTables.generate_table(
    headers=headers, 
    rows=rows, 
    empty_cell_placeholder='No data'
)
print(formatted_table)
```

image

```pythno
colored_table = PrettyTables.generate_table(
    headers=headers, 
    rows=rows, 
    empty_cell_placeholder='No data',
    colors=[TableColors.HEADER, TableColors.OKBLUE, TableColors.FAIL, TableColors.UNDERLINE]
)
print(colored_table)
```

image

```python
truthy_table = PrettyTables.generate_table(
    headers=headers, 
    rows=rows, 
    empty_cell_placeholder='No data',
    truthy=3
)
print(truthy_table)
```

image

```python
truthy_table_custom = PrettyTables.generate_table(
    headers=headers, 
    rows=rows, 
    empty_cell_placeholder='No data',
    colors=[TableColors.OKCYAN, TableColors.WARNING],
    truthy=3
)
print(truthy_table_custom)
```

image

File: README.md
Comment (Justintime50) [inline README.md]: With the addition of conditional coloring, let's update the documentation here.

Can you also include a list/table of all possible colors so people don't have to dive into code to see the options?

File: pretty_tables/tables.py
Comment (Justintime50) [inline pretty_tables/tables.py]: I know it's obnoxious, but can you newline the end of each docblock? This is how I format all my code and I'd like to keep it uniform:

```
"""A class to contain constants for colors for the table."""
```

to

```
"""A class to contain constants for colors for the table.
"""
```

File: pretty_tables/tables.py
Comment (Justintime50) [inline pretty_tables/tables.py]: I had them setup as strings initially because we want to pass strings to the new formatted table. I just want to check that this still occurs by changing them to booleans? I remember having to specifically set them as strings for a reason.

File: pretty_tables/tables.py
Comment (Justintime50) [inline pretty_tables/tables.py]: Please correct `mach` to `match`.

File: pretty_tables/tables.py
Comment (Justintime50) [inline pretty_tables/tables.py:142]: Nice!

File: test/unit/test_tables.py
Comment (Justintime50) [inline test/unit/test_tables.py]: Would it be just as easy to remove the `\` breaks from each line and wrap this in parentheses? This goes for all these tests. Doing so seems a bit cleaner to me. 

eg:
```python
assert pretty_table == (
    '| \x1b[94mID\x1b[0m | \x1b[95mName  \x1b[0m | \x1b[1mOccupation       \x1b[0m '
    '| \x1b[92mEmployed\x1b[0m |\n'
    '| \x1b[94m1 \x1b[0m | \x1b[95mJustin\x1b[0m | \x1b[1mSoftware Engineer\x1b[0m '
    '| \x1b[92mTrue    \x1b[0m |\n'
    '| \x1b[94m2 \x1b[0m | \x1b[95mMisty \x1b[0m | \x1b[1mReceptionist     \x1b[0m '
    '| \x1b[92mFalse   \x1b[0m |\n'
    '| \x1b[94m3 \x1b[0m | \x1b[95mJohn  \x1b[0m | \x1b[1mNo data          \x1b[0m '
    '| \x1b[92mFalse   \x1b[0m |'
)
```

File: test/unit/test_tables.py
Comment (Justintime50) [inline test/unit/test_tables.py:11]: Admittedly this was the quick and dirty approach to just get coverage, had every intention of replacing this with a conftest, so thank you!

**Review by Justintime50 — CHANGES_REQUESTED** [review]
A couple of additional changes needed here before we merge in, a few random syntax/grammar items but especially updating the documentation to reflect the new color feature and its usage. Users should be able to gather all usage info from the README without needing to dive into the code. Thanks!

File: test/unit/test_tables.py
Comment (gagelarsen) [reply test/unit/test_tables.py:11]: No problem

File: pretty_tables/tables.py
Comment (gagelarsen) [reply pretty_tables/tables.py:142]: Figured it would look better ;)

File: pretty_tables/tables.py
Comment (gagelarsen) [reply pretty_tables/tables.py]: Yuck. I can, but is that really the way you like to format it?

I usually do either:
 
```python
“””comment goes here”””
```

Or 

```python
“””
Comment goes here.
“””
```

File: pretty_tables/tables.py
Comment (gagelarsen) [reply pretty_tables/tables.py]: Can you explain what you mean?

File: pretty_tables/tables.py
Comment (Justintime50) [reply pretty_tables/tables.py]: Haha yeah, not the biggest fan either, but it's how we do it at work (due to auto-formatting of xyz blah blah blah.) Literally all my other projects have it setup with the ending `"""` on its own line, even for one-liners.

File: pretty_tables/tables.py
Comment (Justintime50) [reply pretty_tables/tables.py]: For instance, if the user passes a boolean `True` to the table, I want to actually print the string `True` and not just have that field be truthy. Same for `False`, the string `False` should appear.

**Comment by Justintime50** [comment]
@gagelarsen can you rebase against `main`? I added GitHub Actions, you can remove that file from this PR.

File: pretty_tables/tables.py
Comment (Justintime50) [reply pretty_tables/tables.py]: You'll want to rebase against `main` here, I was silly and created this dumb shim initially when really all I needed was to coerce them to a string in the end instead of doing this. This chunk of code has been refactored since.

**Comment by Justintime50** [comment]
@gagelarsen wanna take another look here? If not I can rebase and push to your branch assuming you allow maintainers to edit your PRs

File: test/unit/test_tables.py
Comment (gagelarsen) [reply test/unit/test_tables.py]: Done

File: pretty_tables/tables.py
Comment (gagelarsen) [reply pretty_tables/tables.py]: Done

File: pretty_tables/tables.py
Comment (gagelarsen) [reply pretty_tables/tables.py]: Done

File: README.md
Comment (gagelarsen) [reply README.md]: Done

File: pretty_tables/tables.py
Comment (gagelarsen) [reply pretty_tables/tables.py]: I rebased. I hate rebasing

File: .github/workflows/PrettyTables.yaml
Comment (Justintime50) [inline .github/workflows/PrettyTables.yaml]: Let's remove this file, I already have GitHub Actions setup to handle testing, linting, and releasing

File: pretty_tables/tables.py
Comment (Justintime50) [reply pretty_tables/tables.py]: I don't believe this was rebased properly as the `true/false` logic shouldn't be here anymore. I realized (duh) that I should have just made the boolean a string when printing and I wouldn't need this piece. Let's remove it (unless I'm overlooking a reason to add it back, I assume it's still here in this diff because of a rebase gone wrong) so that we don't reintroduce previous behavior accidentally.

File: pretty_tables/tables.py
Comment (Justintime50) [inline pretty_tables/tables.py]: Is there a better way we can lay this out instead of the 3 nested if statements that seem to be doing similar things? It makes it really difficult to read through the code. If there's not a better way, maybe a simple comment to describe the breakdown of if/else logic here to assist?

Could we also just assert a truth value? `if colors or truthy`, or do we need to specifically be checking for `not None` here?

File: README.md
Comment (Justintime50) [reply README.md]: Let's move this section about `colors` to the bottom of the `usage` section of the README.

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Thanks for looking back into this! It's pretty close, a couple of last things and I think the rebase got off somehow as there is still a bit of the old logic that got put back somehow.

File: .github/workflows/PrettyTables.yaml
Comment (gagelarsen) [reply .github/workflows/PrettyTables.yaml]: Done

File: pretty_tables/tables.py
Comment (gagelarsen) [reply pretty_tables/tables.py]: The logic is a bit strange here, but it is because the colors and truthy are very interelated. I added comments.

File: pretty_tables/tables.py
Comment (Justintime50) [inline pretty_tables/tables.py]: `if colors no colors` typo? Doesn't make sense as written.

File: README.md
Comment (Justintime50) [inline README.md]: Let's kill this leftover whitespace here.

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Two tiny last things and then I'll merge!

File: README.md
Comment (gagelarsen) [reply README.md]: done

Comment (gagelarsen) [reply README.md]: done 'ma lord

**Review by Justintime50 — APPROVED** [review]
Thanks again for the PR! Tests were especially good.

File: pretty_tables/tables.py
Comment (gagelarsen) [reply pretty_tables/tables.py]: Yes, my captain.

**Comment by Justintime50** [comment]
Colors just released in `v1.2.0`!

**Pull request merged**
